addons/mass_mailing/data/mass_mailing_data.xml

```diff
@@ -8,8 +8,8 @@
             <field name="state">code</field>
             <field name="code">model._process_mass_mailing_queue()</field>
             <field name="user_id" ref="base.user_root" />
-            <field name="interval_number">60</field>
-            <field name="interval_type">minutes</field>
+            <field name="interval_number">1</field>
+            <field name="interval_type">days</field>
```

Comment:
That seems unnecessarily risky as third parties may have overridden this cron and rely on it running properly? The cron should do essentially nothing when it has nothing to do (and the table should rather be indexed properly if doing nothing is an excessive cost).

Reply:
Isn't it the whole purpose of cron triggers to lessen number of times a cron is called ?

Reply:
> Isn't it the whole purpose of cron triggers to lessen number of times a cron is called ?

'bit of both? I would expect *that* to mostly an issue for the cases where we need reactivity e.g. crons triggered every minute because the operation they run needs to be extremely reactive. Even more so if determining *whether* there is work to do requires ex(te|pe)nsive work.

Here the cron runs once an hour and does 2 `search`. That doesn't sound like an active area of performance issues?

Reply:
Nope I don't think this is a big issue to run it frequently, just asking. We could also say we change the default internal to have the inverse default behavior: by default consider everything is smooth and let the cron run once every day. And if necessary users may tweak it.

Anyway probably better to left it untouched as it is quite an old cron and indeed, people may use it for other purpose through inheritance. Mass mailing is quite a big app.

Reply:
> (...) Even more so if determining _whether_ there is work to do requires ex(te|pe)nsive work.

It *is* a problem, even when determining whether there is any work to do is cheap. Because just waking up the registry to call the cron method is expensive ;-)

On multi-tenant systems with thousands of databases, this means a cron worker is busy 24/7 (= 1 CPU core wasted), just to keep up with the "let's run this check every hour just in case". Let's say loading a registry takes ~1s, then 3600 databases is the most you can process with 1 worker every hour. And that's just to run methods that do no work - you have no time left for the real work.

Unlocking this bottleneck was a major driver for designing the trigger mechanism. Now we can convert many "queue processing" crons to low-frequency, and trigger them up only when we know there is actual work to do. Suddenly the cron workers' life has a purpose again! :-)

Reply:
But then we're kinda back to the sub-question: why even bother running it daily?

> Of course in practice we have more than 1 such job per database, and some of them are scheduled to run every 5 minutes - imagine what the cron workers do all day long...

No that I completely agree with and specifically mention "high-frequency" crons (as that's pretty much all of the mergebot's).

Reply:
Running it daily is a security, if the trigger mechanism proves to be reliable then maybe we can disable the "automatic" execution and rely only on triggers

Reply:
> But then we're kinda back to the sub-question: why even bother running it daily?

Good point, we don't need to. It's mostly a safety net while we're still in the early days of the trigger mechanism, still somewhat of a WIP. We could later switch to monthly frequency, or even introduce an infinite interval (`nextcall` set to `NULL` after execution?) where it only runs when triggered.

Reply:
Or use it to cleanup traces on a monthly basis :D .

Reply:
fair 'nuf